mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -561,6 +588,177 @@ class WidgetAdapter(
         }
     }
 
+    class InputViewHolder internal constructor(initData: ViewHolderInitData) : LabeledItemBaseViewHolder(
+        initData,
+        R.layout.widgetlist_inputitem,
+        R.layout.widgetlist_inputitem_compact
+    ) {
+        private val inputTextLayout: TextInputLayout = itemView.findViewById(R.id.widgetinput)
+        private val inputText: TextInputEditText = itemView.findViewById(R.id.widgetinputvalue)
+        private var isBinding = false
+        private var hasChanged = false
+
+        private var updateJob: Job? = null
+        private var oldValue: String? = null
+
+        init {
+            inputText.doAfterTextChanged { if (!isBinding) hasChanged = true }
+            inputText.setOnFocusChangeListener { view, hasFocus ->
+                if (!hasFocus) {
+                    hideKeyboard(view)
+                    if (hasChanged) updateValue()
+                }
+            }
+            inputText.setOnEditorActionListener { view, action, _ ->
+                if (action == EditorInfo.IME_ACTION_DONE) {
+                    hideKeyboard(view)
+                    if (hasChanged) updateValue()
```

Comment:
Nit: Could you please add braces for if and else blocks? I know the Kotlin code style doesn't strictly require them, but (at least) I prefer using them for the additional clarity added by new line + indention.

Reply:
done